<?xml version="1.0"?>
<launch>
  <!-- <include file="$(find pid)/launch/servo_sim.launch"/> -->
  <!-- <node pkg="tf" type="static_transform_publisher" name="base_link_to_imu" args="0 0 0 0 0 0 1 gnss_link imu_link 1000" />
  <node pkg="tf" type="static_transform_publisher" name="base_link_to_lidar" args="0 0 2.3 0 0 0 1 base_link lidar_link 1000" />
  <node pkg="tf" type="static_transform_publisher" name="base_link_to_gnss" args="-0.2 0 2.2 0 0 0 1 base_link gnss_link 1000" />
  <node pkg="tf" type="static_transform_publisher" name="tmp_map_to_gnss" args="0.0 0 0.0 0 0 0 1 map gnss_link 100" /> -->


  <node name="highspeed_ctrl" pkg="highspeed_ctrl" type="highspeed_ctrl" output="screen">      
        <remap from="/vesc/ackermann_cmd" to="/vesc/low_level/ackermann_cmd_mux/input/nav_hmcl"/>

        
      <param name="waypoint_topic" value="/track_info"/> 

      <param name="Pminimum_lookahead_distance" value="0.8"/> 
      <param name="Pmaximum_lookahead_distance" value="2.5"/> 
      <param name="Pspeed_to_lookahead_ratio" value="0.6"/> 
      
      <param name="Pemergency_stop_distance" value="0.0"/> 
      <param name="Pspeed_thres_traveling_direction" value="0.0"/> 
      <param name="Pmax_acceleration" value="50.0"/> 
      <param name="Pdistance_front_rear_wheel" value="0.33"/> 

      <param name="vel_lookahead_ratio" value="0.5"/>       
      <param name="speed_minimum_lookahead_distance" value="0.0"/> 
      <param name="speed_maximum_lookahead_distance" value="2.5"/> 

      <param name="x_vel_filter_cutoff" value="10.0"/> 
      <param name="y_vel_filter_cutoff" value="8.0"/> 

      <param name="odom_topic" value="/pose_odom_virtual"/> 


      <param name="path_smoothing_times_" value="1"/> 
      <param name="curvature_smoothing_num_35" value="35"/> 
      <param name="path_filter_moving_ave_num_" value="1"/> 
      <param name="lookahead_path_length" value="10.0"/> 
      <param name="curv_lookahead_path_length" value="10.0"/> 

      
      
      <param name="wheelbase" value="0.33"/> 
      <param name="lf" value="0.165"/> 
      <param name="lr" value="0.165"/> 
      <param name="mass" value="2.0"/> 
      <param name="dt" value="0.05"/> 



    
  </node>
  
</launch>
